tutorials/image/cifar10/cifar10_multi_gpu_train.py

```diff
@@ -162,13 +162,14 @@ def train():
 
     # Calculate the gradients for each model tower.
     tower_grads = []
-    for i in xrange(FLAGS.num_gpus):
-      with tf.device('/gpu:%d' % i):
-        with tf.name_scope('%s_%d' % (cifar10.TOWER_NAME, i)) as scope:
-          # Calculate the loss for one tower of the CIFAR model. This function
-          # constructs the entire CIFAR model but shares the variables across
-          # all towers.
-          loss = tower_loss(scope)
+    with tf.variable_scope(tf.get_variable_scope()):
+      for i in xrange(FLAGS.num_gpus):
+        with tf.device('/gpu:%d' % i):
+          with tf.name_scope('%s_%d' % (cifar10.TOWER_NAME, i)) as scope:
+            # Calculate the loss for one tower of the CIFAR model. This function
+            # constructs the entire CIFAR model but shares the variables across
+            # all towers.
+            loss = tower_loss(scope)
 
           # Reuse variables for the next tower.
           tf.get_variable_scope().reuse_variables()
```

Comment:
Hi @nealwu, I will explain. The new operations created in `tower_loss()` (for each tower) should be definitely scoped with the proper prefix (e.g. `TOWER_0`). For the rest of operations, I thought they are not relevant to *op creation*, so basically they will be not affected by namespace.

However, I agree with you that I will have to include those lines in the scope. It agrees to the previous version of code, and we can get rid of any potential side-effects (e.g. new op might be created due to the future changes). It will look more natural. I will update the changeset. Thanks!